core/src/main/java/org/jdbi/v3/StatementContext.java

```diff
@@ -61,7 +63,15 @@
      * @param type the target type to map to
      * @return a ResultColumnMapper for the given type, or null if no column mapper is registered for the given type.
      */
-    <T> ResultColumnMapper<? extends T> columnMapperFor(Class<T> type);
+    ResultColumnMapper<?> columnMapperFor(Type type);
+
+    /**
+     * Obtain an argument for given value in this context
+     * @param type the type of the argument.
+     * @param value the argument value.
+     * @return an Argument for the given value, or null if no argument factory is registered for the given type
```

Comment:
Will this actually ever return null?  `BuiltInArgumentFactory` will in the worst case go with `PreparedStatement.setObject`...